### PR TITLE
fix(buffer): do not filter help buffer

### LINKED ISSUE #26650: ":colorscheme vim" doesn't properly revert to the old legacy color scheme
### Problem

The documentation under `:help news` has the following:
"Use :colorscheme vim to revert to the old legacy color scheme."

Using this colorscheme results in colors that don't match the old legacy color scheme.

### Steps to reproduce

```vim
nvim --clean
" set termguicolors (or notermguicolors) so it's set consistently for experiments
:set termguicolors
" the following should be omitted for nvim<0.10
:colorscheme vim
" call :help to show text
:help
:only
```

### Expected behavior

The side-by-sides below show Neovim v0.10 (left) versus Neovim 0.9.4 (right). For the top screenshots, `termguicolors` was set. For the bottom screenshots, `notermguicolors` was set. I expected the left side colors would match the right side colors.

### `termguicolors`

termguicolors

### `notermguicolors`

notermguicolors

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1880+ge164f4c27

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Kubuntu 23.10

### Terminal name/version

Konsole 23.08.1

### $TERM environment variable

xterm-256color

### Installation

build from repo

### PR DESCRIPTION
**Pull request by clason**
Problem: If a help buffer is opened without legacy syntax set (because treesitter is enabled), Vim strips (some) markup. This means the syntax engine fails to parse (some) syntax if treesitter highlighting is disabled again.

Solution: Do not strip the help buffer of markup since (legacy or treesitter) highlighting is always enabled in Nvim. Similarly, remove redundant setting of filetype and give the function a more descriptive name.

fixes #26650

### CODE REVIEW

**Comment by clason** [comment]
(nice sleuthing, @seandewar !)

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -302,10 +302,6 @@ int open_buffer(int read_stdin, exarg_T *eap, int flags_arg)
     }
 #endif
```
Comment (zeertzjq) [inline src/nvim/buffer.c:304]:
```suggestion
```

**Review by zeertzjq — APPROVED** [review]

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/help.c [hunk 2]
```diff
@@ -653,35 +652,6 @@ void fix_help_buffer(void)
     curbuf->b_ro_locked--;
   }
```
Comment (clason) [inline src/nvim/help.c:653]: What does this part even do? According to my tests, help buffers are detected without it just fine (and doesn't this code only run on help buffers, so filetype should already be set)?

File: src/nvim/help.c [hunk 3]
```diff
@@ -653,35 +652,6 @@ void fix_help_buffer(void)
...
-        } else if (line[len - 1] == '~') {
-          // blank-out a '~' at the end of line (header marker)
-          line = ml_get_buf_mut(curbuf, lnum);
-          line[len - 1] = ' ';
-        }
-      }
-    }
-  }
-
   // In the "help.txt" and "help.abx" file, add the locally added help
```
Comment (clason) [inline src/nvim/help.c:655]: Do we even have `help.abx` files? The longer I look at this function, the more I want to burn it down...

Comment (zeertzjq) [reply src/nvim/help.c:655]: Plugin translations

File: src/nvim/help.c (hunk 2 above)
Comment (clason) [reply src/nvim/help.c:653]: Is this because Vim (unlike Neovim) may not run filetype by default?

File: src/nvim/help.c (hunk 3 above)
Comment (clason) [reply src/nvim/help.c:655]: Oh, it's for translations....

**Pull request merged**